tmva/sofie/inc/TMVA/SOFIE_common.hxx

```diff
@@ -104,9 +106,15 @@ ETensorType GetTemplatedType(T /*obj*/ ){
 }
 
 namespace UTILITY{
-template<typename T>
-T* Unidirectional_broadcast(const T* original_data, const std::vector<size_t> original_shape, const std::vector<size_t> target_shape);
-std::vector<size_t> Multidirectional_broadcast(std::vector<size_t> input1_shape, std::vector<size_t> input2_shape);
+// Check if two shapes are equal
+bool AreSameShape(const std::vector<size_t>& /*shapeA*/, const std::vector<size_t>& /*shapeB*/);
+
+// Multidirectional broadcast a list of tensors to the same shape
```

Comment:
It would be better to pass shape as a const reference:
const std::vector<std::vector<size_t> &

Similar for the other function below

Also since it is a declaration we don't need to add a C++ comment `/* */` on the passed argument names

Reply:
For UnidirectionalBroadcast and MultidirectionalBroadcast the shape is prepended with ones if its size is less than the size of the common shape. I think its better to take the shapes by value and modify them if its necessary.